src/main/java/com/google/googleidentity/oauth2/client/InMemoryClientDetailsService.java

```diff
@@ -0,0 +1,77 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.client;
+
+
+import com.google.googleidentity.user.UserDetails;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Default InMemory ClientDetailsService for client information Store
+ * An Implementation for {@link ClientDetailsService}
+ */
+
+public class InMemoryClientDetailsService implements ClientDetailsService{
+
+    private ConcurrentHashMap<String, ClientDetails.Client> clientStore
+            = new ConcurrentHashMap<String, ClientDetails.Client>();
+
+    @Override
+    public ClientDetails.Client getUserByID(String clientID) {
+        return clientStore.getOrDefault(clientID, null);
+    }
+
+    @Override
+    public boolean updateClient(ClientDetails.Client client) {
+        if (client == null) {
```

Comment:
If client is allowed to be null, then use optional, if client shall not be null, then throw exception, you can use https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html

Same for other null checks.

Reply:
I changed in some places, but optional type can not be used in @SessionScoped class since it is not Serilizable Object.